doc/development/plugins.rst

```diff
@@ -101,15 +101,35 @@ as well as potential further capabilities, by providing the following class attr
       conversion between the two conventions takes place automatically
       by the plugin device.
 
-* :attr:`.Device._capabilities`: a dictionary containing information about the capabilities of
-  the device. Keys currently supported include:
+* :func:`.Device.capabilities`: A class method which returns the dictionary of capabilities of a device. A
+  new device should override this method to retrieve the parent classes' capabilities dictionary, make a copy
+  and update and/or add capabilities before returning the copy.
+
+  Examples of capabilities are:
 
   * ``'model'`` (*str*): either ``'qubit'`` or ``'CV'``.
 
-  * ``'inverse_operations'`` (*bool*): ``True`` if the device supports
+  * ``'supports_exact'`` (*bool*): ``True`` if the device is a simulator that returns analytic results.
+
+  *  ``'supports_sampled'`` (*bool*): ``True`` if outputs are estimates computed from samples of measurement outcomes.
+
+  * ``'supports_inverse_operations'`` (*bool*): ``True`` if the device supports
     applying the inverse of operations. Operations which should be inverted
     have the property ``operation.inverse == True``.
 
+  *  ``'supports_tensor_observables'`` (*bool*): ``True`` if the device supports observables composed from tensor
+     products such as ``PauliZ(wires=0) @ PauliZ(wires=1)``.
+
+  *  ``'executes_in_remote'`` (*bool*): ``True`` if the device runs computations on remote servers or devices.
+
+  *  ``'takes_fixed_number_of_wires'`` (*bool*): ``True`` if the device's number of wires are fixed.
+
+  *  ``'runs_noisy_computations'`` (*bool*): ``True`` if the device's results are stochastic due to
+     physical or simulated noise.
```

Comment:
> if the device's results are stochastic due to physical

Would that mean that hardware devices would have this attribute be set to `True`?

Reply:
Yes, and some simulators...